### ISSUE TITLE
3.6.0: Incorrect props validation in constructor

### ISSUE THREAD

**Issue by necolas** [yannickcr/eslint-plugin-react#254]
[Source code](https://github.com/necolas/react-native-web/blob/1c444569ae8f12130197459ed0d34829826008a0/src/components/Image/index.js#L60):

```js
class Image extends React.Component {
  constructor(props, context) {
    super(props, context)
    this.state = { status: props.source.uri ? STATUS_PENDING : STATUS_IDLE }
  }
}
```

As you can see, `uri` is not a prop. But after updating to `3.6.0` it is producing a [linting error](https://travis-ci.org/necolas/react-native-web/builds/86059160)

```
/home/travis/build/necolas/react-native-web/src/components/Image/index.js
  60:34  error  'uri' is missing in props validation for Image  react/prop-types
```

The issue is not reproduced if the source is changed to:

```js
class Image extends React.Component {
  constructor(props, context) {
    super(props, context)
    const { uri } = props.source
    this.state = { status: uri ? STATUS_PENDING : STATUS_IDLE }
  }
}
```

**Comment by yannickcr** [comment]
Thanks for the report, I'll have a look.
